src/Symfony/Component/Security/Http/EventListener/IsGrantedAttributeListener.php

```diff
@@ -39,7 +39,16 @@ public function __construct(
 
     public function onKernelControllerArguments(ControllerArgumentsEvent $event): void
     {
-        if (!$attributes = $event->getAttributes(IsGranted::class)) {
+        $attributes = [];
+        foreach ($event->getAttributes() as $attribute) {
+            foreach ($attribute as $attribute) {
+                if ($attribute instanceof IsGranted) {
+                    $attributes[] = $attribute;
+                }
+            }
```

Comment:
Maybe something like this would be enough?

```suggestion
            if (isset($attribute[0]) && $attribute[0] instanceof IsGranted) {
                $attributes = $attribute;
            }
```

Reply:
I updated using another style - not sure we should optimize for only one attribute